salt/client/__init__.py

```diff
@@ -1698,6 +1699,33 @@ def _prep_pub(self,
 
         return payload_kwargs
 
+    def _prep_jid(self, jid=None, nocache=False, subscribe=False):
+        '''
+        Attempt to generate a jid for this publication if necessary
+        '''
+         # remove empty string
+        jid = jid or None
+
+        # Retrieve the jid
+        fstr = '{0}.prep_jid'.format(self.opts['master_job_cache'])
+        try:
+            # Retrieve the jid
+            jid = self.returners[fstr](nocache=nocache, passed_jid=jid)
+        except Exception:
+            # The returner is not present
```

Comment:
I should also have mentioned this on the initial review, don't use string formatting, `{}` in log calls, use string substitution, `log.info('I have a %s to say', 'thing')`

Reply:
fixed this - updated in master.py as well